webapp/shop/cred/views.py

```diff
@@ -1391,6 +1392,10 @@ def cred_faq(
 
 @shop_decorator(area="cred", permission="user", response="html")
 def cred_submit_form(**_):
+    ability_screen_id = flask.request.args.get("exam_id")
+    exam_name = EXIT_SURVEY_ABILITY_SCREEN_ID_MAPPING.get(
+        ability_screen_id, None
+    )
     if flask.request.method == "GET":
         return flask.render_template("credentials/exit-survey.html")
```

Comment:
Should we move the exam extraction logic below the early exit?

```suggestion
    if flask.request.method == "GET":
        return flask.render_template("credentials/exit-survey.html")
    
    ability_screen_id = flask.request.args.get("exam_id")
    exam_name = EXIT_SURVEY_ABILITY_SCREEN_ID_MAPPING.get(
        ability_screen_id, None
    )
```

Reply:
I updated this code, the extraction from url is required **before** the GET method because the exam_id needs to be passed down to the form template.